### PR TITLE
Optional decoders

### PR DESCRIPTION
**Pull request by chyvonomys**
This PR adds ability to turn off unwanted decoders and their dependencies. (`wav/hound`, `vorbis/lewton`, `flac/claxon`). Can be useful if you use `rodio` with one format only and don't need any extra. It is also valid to turn off all decoding capabilities. By default all three are turned on.
(Similar thing is possible in `image` crate, I used it there and thought it may be useful for `rodio` as well)

### CODE REVIEW

File: src/decoder/mod.rs [hunk 1]
```diff
@@ -30,29 +36,36 @@ impl<R> Decoder<R>
     /// Builds a new decoder.
     ///
     /// Attempts to automatically detect the format of the source of data.
-    pub fn new(data: R) -> Result<Decoder<R>, DecoderError> {
-        let data = match wav::WavDecoder::new(data) {
-            Err(data) => data,
+    pub fn new(_data: R) -> Result<Decoder<R>, DecoderError> {
+        #[cfg(feature = "wav")]
```
Comment (tomaka) [inline src/decoder/mod.rs]: I'd prefer if the variable names didn't have a _ prefix, and you just put a `#![allow(unused_variables)]` in the function instead.

File: src/lib.rs [hunk 2]
```diff
@@ -108,13 +113,15 @@ mod sink;
...
 pub mod buffer;
+#[cfg(any(feature = "wav", feature = "flac", feature = "vorbis"))]
 pub mod decoder;
 pub mod dynamic_mixer;
 pub mod queue;
 pub mod source;
 
 /// Plays a sound once. Returns a `Sink` that can be used to control the sound.
 #[inline]
+#[cfg(any(feature = "wav", feature = "flac", feature = "vorbis"))]
```
Comment (tomaka) [inline src/lib.rs]: It's not great to remove this function if you don't enable any decoder. I think the `DecoderError` should just be exposed in any situation instead.

**Comment by chyvonomys** [comment]
Yea, I agree. I made necessary changes to have `play_once` and `Decoder` in place.
Tested on `play_wav` example. It panics with `UnrecognisedFormat` as expected now.

**Comment by tomaka** [comment]
Is the split between two different versions of `Decoder` necessary?
I feel like it should just work with the empty enum.

**Comment by chyvonomys** [comment]
Yes, otherwise compiler complains about unused `R`.
Other option would be to have one more variant for `DecoderImpl`, but I don't feel like that is better.

**Comment by tomaka** [comment]
I see, thanks!

**Pull request merged**